apps/src/javalab/CommitDialog.jsx

```diff
@@ -2,49 +2,58 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import i18n from '@cdo/javalab/locale';
 import color from '@cdo/apps/util/color';
-import {CompileStatus} from './constants';
 import StylizedBaseDialog, {
   FooterButton
 } from '@cdo/apps/componentLibrary/StylizedBaseDialog';
+import {connect} from 'react-redux';
 
-export default class CommitDialog extends React.Component {
+class CommitDialog extends React.Component {
   state = {
     filesToBackpack: [],
-    commitNotes: null,
-    showCompileStatus: false
+    commitNotes: '',
+    saveInProgress: false,
+    hasSaveError: false
   };
 
   renderFooter = () => {
-    let compileStatusContent = '';
+    let footerIcon = '';
+    let footerMessageTitle = '';
+    let footerMessageText = '';
     let commitText = i18n.commit();
-    const isCommitButtonDisabled = !this.state.commitNotes;
+    const isCommitButtonDisabled =
+      !this.state.commitNotes || this.state.saveInProgress;
     if (this.state.filesToBackpack.length > 0) {
       commitText = i18n.commitAndSave();
     }
-    if (this.state.showCompileStatus) {
-      switch (this.props.compileStatus) {
-        case CompileStatus.LOADING:
-          compileStatusContent = i18n.compiling();
-          break;
-        case CompileStatus.ERROR:
-          compileStatusContent = i18n.compileFailed();
-          break;
-      }
+
+    // TODO: Add compile status here
+    if (this.state.saveInProgress) {
+      footerIcon = (
+        <span className="fa fa-spin fa-spinner" style={styles.spinner} />
+      );
+      footerMessageTitle = i18n.saving();
+    } else if (this.state.hasSaveError) {
+      footerIcon = (
+        <span className="fa fa-exclamation-circle" style={styles.iconError} />
+      );
+      footerMessageTitle = i18n.backpackSaveErrorTitle();
+      footerMessageText = i18n.backpackSaveErrorMessage();
     }
 
     return [
-      <div
-        key="compile-status"
-        style={{...styles.bold, ...styles.compileStatus}}
-      >
-        {compileStatusContent}
+      <div key="footer-status" style={styles.footerStatus}>
+        <div style={styles.footerIcon}>{footerIcon}</div>
+        <div style={styles.footerMessage}>
+          <div style={styles.footerMessageTitle}>{footerMessageTitle}</div>
+          <div style={styles.footerMessageText}>{footerMessageText}</div>
```

Comment:
For accessibility purposes, the title should be a header element and the text a p element. Or if there are no other header elements on the dialog, both should be p elements. For the header, you could alternatively use an [aria role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/heading_role).

Reply:
Long read: https://developer.mozilla.org/en-US/docs/Learn/Accessibility/HTML 
Short read: https://blog.segunolalive.com/posts/improving-accessibility-with-div-and-span/#heading-when-to-use-a-div-or-span
(If interested)

Reply:
Aaaand I'm realizing this is a bit more nuanced. Since this is a short, one-line message, it doesn't really matter. 
Short discussion that seems to summarize it: https://stackoverflow.com/questions/67792979/does-putting-text-directly-in-a-div-have-accessibility-implications